src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunner.java

```diff
@@ -0,0 +1,92 @@
+package org.jetbrains.bsp.bazel.server.bazel;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.bsp.bazel.server.bazel.data.BazelProcessResult;
+
+public class BazelRunner {
+
+  private static final Logger LOGGER = LogManager.getLogger(BazelRunner.class);
+
+  private static final String PUBLISH_ALL_ACTIONS = "--build_event_publish_all_actions";
+  private static final String BES_BACKEND = "--bes_backend=grpc://localhost:";
+
+  private final String bazel;
+
+  private Optional<Integer> besBackendPort = Optional.empty();
+
+  public BazelRunner(String bazelBinaryPath) {
+    this.bazel = bazelBinaryPath;
+  }
+
+  public BazelRunnerCommandBuilder commandBuilder() {
+    return new BazelRunnerCommandBuilder(this);
+  }
+
+  BazelProcessResult runBazelCommandBes(
+      String command, List<String> flags, List<String> arguments) {
+
+    List<String> newFlags = getBesFlags(flags);
+    return runBazelCommand(command, newFlags, arguments);
+  }
+
+  private List<String> getBesFlags(List<String> flags) {
+    List<String> newFlags = new ArrayList<>();
```

Comment:
Actually here too you can use ``Lists.newArrayList``.

Reply:
done